### PR TITLE
grep exit code 1 is not an error. It just means, there was no matching pattern found.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <shishir.mahajan@redhat.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
Let's say I have a regular volume `foobar` and I call `isThinlyProvisioned()` function to determine if `foobar` is a regular or thin volume.

`IsThinlyProvisioned()` will do a `lvdisplay /dev/volume_group_one/foobar | grep "LV Pool"` where `volume_group_one` is my volume group name. Since foobar is a regular volume, grep will return no output and an `exit code 1`.

http://www.gnu.org/software/grep/manual/html_node/Exit-Status.html
According to GNU Grep:

Exit code 0: Match found
Exit code 1: No match found
Exit code 2: Error

With Exit code 1, our golang code will treat this as an error and https://github.com/projectatomic/docker-lvm-plugin/blob/master/driver.go#L320 will return an error.
We need to ignore this error, since this is not really an error. It is just a pattern match not found.

More Explanation here:
https://github.com/projectatomic/docker-lvm-plugin/pull/9#issuecomment-235129127

Shishir

**Comment by shishir-a412ed** [comment]
@rhatdan @runcom Please review.

File: driver.go [hunk 1]
```diff
@@ -318,7 +318,9 @@ func isThinlyProvisioned(vgName, lvName string) (bool, error) {
 	}
 	w.Close()
 	if err := cmd2.Wait(); err != nil {
-		return false, err
+		if err.Error() != "exit status 1" {
```
Comment (runcom) [inline driver.go]: checking strings in Golang is generally bad and I'm not sure any grep implementation in distros is returning `exit status 1`, so I'd say, try with:

https://github.com/docker/docker/blob/master/pkg/integration/utils.go#L159

which calls `ProcessExitCode(err error)` here:

https://github.com/docker/docker/blob/master/pkg/integration/utils.go#L35

**Comment by shishir-a412ed** [comment]
@runcom PTAL.

Shishir

**Comment by runcom** [comment]
Seems ok to me Shishir

**Pull request merged**